networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,305 @@
+
+    DomiRank tends to have only positive values for relatively low
+    competition levels (small), however as the competition
+    levels increase, negative values might emerge. Nodes with negative
+    dominance represent completely submissive nodes, which instead
+    of fighting for their resources/dominance, directly give up these
+    resources to their neighborhood.
+
+    Finally, DomiRank centrality does not require the network to be weakly or strongly
+    connected, and can be applied to networks with many components.
+
+    Parameters
+    ----------
+    G : graph
+        A NetworkX graph.
+
+    analytical: bool, optional (default=False)
+        whether the analytical or iterative formulation
+        for computing DomiRank should be used.
+        It is recommended that ''analytical=False'' for
+        large networks.
+
+    sigma: float, optional (default=0.95)
+        The level of competition for DomiRank. Must satisfy
+        that $\sigma \in (0,1)$.
+
+    dt: float, optional (default=0.1)
+        The step size for the Newton iteration.
+
+    epsilon: float, optional (default=1e-5)
+        The relative stopping criterion for convergence.
+
+    max_iter: integer, optional (default=50)
+        Maximum number of Newton iterations allowed.
+        It is recommended that ''max_iter >= 50''.
+
+    patience: integer, optional (default=10)
+        The number of steps between convergence checks.
+        It is recommended that ''patience >= 10''.
+
+
+    Returns
+    -------
+    nodes : dictionary
+       Dictionary keyed by node with DomiRank centrality of the node as value.
+
+    sigma : float
+       $\sigma$ normalized by the smallest eigenvalue.
+
+    converged: boolean
+       Whether the centrality computation converged.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(5)
+    >>> centrality, sigma, converged = nx.domirank(G)
+    >>> print([f"{node} {centrality[node]:0.2f}" for node in centrality])
+    ['0 -0.54', '1 1.98', '2 -1.08', '3 1.98', '4 -0.54']
+
+    Raises
+    ------
+    NetworkXPointlessConcept
+        If the graph `G` is the null graph.
+
+    NetworkXUnfeasible
+        If sigma is negative (and thus outside its bounds): ``sigma < 0``.
+
+    NetworkXUnfeasible
+        If ``sigma > 1`` with the ``analytical=False`` argument.
+
+    Warning
+        If supercharging the competition parameter for the analytical solution: ``sigma > 1``.
+
+    Warning
+        If one is using the analytical solution for a large system, i.e. ``len(G) > 5000``, as the algorithm will be slow.
+
+    See Also
+    --------
+    :func:`~networkx.algorithms.centrality.degree_centrality`
+    :func:`~networkx.algorithms.link_analysis.pagerank_alg.pagerank`
+    :func:`~networkx.algorithms.link_analysis.hits_alg.hits`
+
+    References
+    ----------
+    .. [1] Engsig, M., Tejedor, A., Moreno, Y. et al.
+    "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
```

Comment:
This line is supposed to be indented, and that's why sphinx [is complaining](https://stackoverflow.com/questions/38761546/markup-ending-without-empty-line-but-markup-is-only-thing-in-comment).
```suggestion
        "DomiRank Centrality reveals structural fragility of complex networks via node dominance."
```